olive/platform_sdk/qualcomm/env.py

```diff
@@ -0,0 +1,103 @@
+# -------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+# --------------------------------------------------------------------------
+
+import os
+import platform
+from pathlib import Path
+
+from olive.platform_sdk.qualcomm.constants import SDKTargetDevice
+
+
+class SDKEnv:
+    def __init__(self, sdk: str, root_env_name: str, target_arch: str = None, dev: bool = False):
+        self.sdk = sdk
+
+        self.root_env_name = root_env_name
+        if not self.root_env_name:
+            raise ValueError("root_env_name is required.")
+
+        self.sdk_root_path = os.environ.get(root_env_name)
+        if not self.sdk_root_path:
+            raise ValueError(f"{root_env_name} is not set.")
+        elif not Path(self.sdk_root_path).exists():
+            raise FileNotFoundError(f"The path {self.sdk_root_path} for {self.root_env_name} does not exist.")
+
+        # dev: Whether to use the development version of the SDK,
+        # only applicable to SNPE/QNN x86_64 linux/windows
+        self.dev = dev
+        self.target_arch = self._verify_target_arch(target_arch) if target_arch else self._infer_target_arch()
+
+    def _verify_target_arch(self, target_device: SDKTargetDevice) -> str:
+        archs = list(Path(self.sdk_root_path).glob(f"lib/{target_device}*"))
+        if len(archs) == 0:
+            raise FileNotFoundError(f"{self.sdk_root_path} missing {target_device}")
+        return archs[0].name
+
+    def _infer_target_arch(self, fail_on_unsupported: bool = True) -> str:
+        """Infer the target architecture from the SDK root path based on platform and processor."""
+        system = platform.system()
+        target_arch = None
+        if system == "Linux":
+            machine = platform.machine()
+            if machine == "x86_64":
+                target_arch = SDKTargetDevice.x86_64_linux
+            else:
+                if fail_on_unsupported:
+                    raise ValueError(f"Unsupported machine {machine} on system {system}")
+        elif system == "Windows":
+            processor_identifier = os.environ.get("PROCESSOR_IDENTIFIER", "")
+            if "ARM" in processor_identifier:
+                target_arch = SDKTargetDevice.arm64x_windows
+            elif "AARCH" in processor_identifier:
+                target_arch = SDKTargetDevice.aarch64_windows
+            else:
+                target_arch = SDKTargetDevice.x86_64_windows
+        else:
+            if fail_on_unsupported:
+                raise ValueError(f"Unsupported system {system}")
+        try:
+            target_arch = self._verify_target_arch(target_arch)
+        except FileNotFoundError:
+            if fail_on_unsupported:
+                raise FileNotFoundError(f"Cannot find target_arch {target_arch} under {self.sdk_root_path}") from None
+            else:
```

Comment:
raise is enough?

Reply:
![image](https://github.com/microsoft/Olive/assets/13343117/9f08afc2-dbe0-4819-b20f-69f07337896f)
Linter suggest to `raise ... from none/err` under `except` clause.

Reply:
https://github.com/microsoft/Olive/blob/main/olive/engine/engine.py#L357

Reply:
just raise it.